wfc_core/src/lib.rs

```diff
@@ -103,24 +100,23 @@ impl fmt::Display for WorldStatus {
     }
 }
 
+#[derive(Clone)]
 pub struct World {
     dims: [u16; 3],
     adjacencies: Vec<Adjacency>,
-    use_shannon_entropy: bool,
 
     slots: Vec<BitVec>,
+    slot_module_weights: Vec<f32>,
```

Comment:
Make this `Option<Vec<f32>>`, because the data will be huge (4 bytes * slot count * module count). We don't want people to pay for things they don't use, especially since we are considering increasing the module count limit soon. The option can also encode the fact that custom weights have been set.